plugins/confluence-plugin/src/components/PageContent.tsx

```diff
@@ -1,138 +1,137 @@
 import type React from "react";
-import { useState, useEffect } from "react";
-import { isEmpty, isNil } from "lodash";
-import "../baseStyles.css";
-import {
-  Box,
-  Text,
-  usePluginContext,
-} from "@cortexapps/plugin-core/components";
-import { getEntityYaml } from "../api/Cortex";
-import { getConfluenceDetailsFromEntity } from "../lib/parseEntity";
+import { useState, useEffect, useMemo } from "react";
+import { isNil } from "lodash";
+import { usePluginContext } from "@cortexapps/plugin-core/components";
+import { Heading, Box } from "@chakra-ui/react";
 
 import Instructions from "./Instructions";
+import Loading from "./Loading";
+import Notice from "./Notice";
+import PageSelector from "./PageSelector";
+
+import {
+  usePluginConfig,
+  useCortexEntityDefinition,
+  useConfluencePageContent,
+} from "../hooks";
+
+import "../baseStyles.css";
 
 const PageContent: React.FC = () => {
-  const [pageContent, setPageContent] = useState<string | undefined>();
-  const [pageTitle, setPageTitle] = useState<string | undefined>();
-  const [entityPage, setEntityPage] = useState<any | string>();
-  const [baseConfluenceUrl, setBaseConfluenceUrl] = useState<string>("");
+  const [entityPage, setEntityPage] = useState<EntityPageI>();
   const [errorStr, setErrorStr] = useState<string>("");
 
   const context = usePluginContext();
 
+  const { isLoading: isPluginConfigLoading, pluginConfig } = usePluginConfig(
+    "confluence-plugin-config"
+  );
+
+  const { isLoading: isEntityDefinitionLoading, entityDefinition } =
+    useCortexEntityDefinition(context.entity?.tag ?? "");
+
+  const baseConfluenceUrl = useMemo(
+    () => pluginConfig?.info?.["x-cortex-definition"]?.["confluence-url"] ?? "",
+    [pluginConfig]
+  );
+
+  const entityPages = useMemo((): EntityPageI[] => {
+    if (!entityDefinition) return [];
+    const entityPages: EntityPageI[] = [];
+    if (Array.isArray(entityDefinition.info?.["x-cortex-confluence"]?.pages)) {
```

Comment:
And some spacing for readability: 

```suggestion

    const entityPages: EntityPageI[] = [];
    if (Array.isArray(entityDefinition.info?.["x-cortex-confluence"]?.pages)) {
```